Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/keydownload/ui/KeyDownloadTestFragmentViewModel.kt

```diff
@@ -0,0 +1,55 @@
+package de.rki.coronawarnapp.test.keydownload.ui
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.asLiveData
+import com.squareup.inject.assisted.AssistedInject
+import de.rki.coronawarnapp.appconfig.AppConfigProvider
+import de.rki.coronawarnapp.diagnosiskeys.download.KeyPackageSyncSettings
+import de.rki.coronawarnapp.diagnosiskeys.download.KeyPackageSyncTool
+import de.rki.coronawarnapp.diagnosiskeys.storage.KeyCacheRepository
+import de.rki.coronawarnapp.util.coroutine.DispatcherProvider
+import de.rki.coronawarnapp.util.network.NetworkStateProvider
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModel
+import de.rki.coronawarnapp.util.viewmodel.SimpleCWAViewModelFactory
+import kotlinx.coroutines.flow.map
+
+class KeyDownloadTestFragmentViewModel @AssistedInject constructor(
+    dispatcherProvider: DispatcherProvider,
+    private val syncSettings: KeyPackageSyncSettings,
+    private val appConfigProvider: AppConfigProvider,
+    private val networkStateProvider: NetworkStateProvider,
+    private val keyPackageSyncTool: KeyPackageSyncTool,
+    private val keyCacheRepository: KeyCacheRepository
+) : CWAViewModel(dispatcherProvider = dispatcherProvider) {
+
+    val isMeteredConnection = networkStateProvider.networkState
+        .map { it.isMeteredConnection }
+        .asLiveData()
+
+    val allowMeteredConnections = syncSettings.allowMeteredConnections.flow.asLiveData()
+
+    val isSyncRunning = MutableLiveData(false)
+
+    fun toggleAllowMeteredConnections() {
+        syncSettings.allowMeteredConnections.update { !it }
+    }
+
+    fun download() {
+        isSyncRunning.postValue(true)
+        launch {
+            keyPackageSyncTool.syncKeyFiles()
+            isSyncRunning.postValue(false)
+        }
+    }
+
+    fun clearDownloads() {
```

Comment:
DRY - funs are nearly the same...

Reply:
Not sure what you mean here?

Reply:
Dry means, Don't repeat yourself (codestyle).. as download and clearDownloads is nearly the same the variable part could be passed as parameter. But its just style, not an issue. So feel free

Reply:
https://github.com/corona-warn-app/cwa-app-android/pull/1550/commits/982ad98b82cf056d4afc30597de53750584e4e68